### PR TITLE
ENH: signal: add `lanczos` and `kaiser_bessel_derived` windows

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref https://github.com/cupy/cupy/issues/7404

Both windows are trivial enough so that just copy pasting scipy implementations seems enough.

### CODE REVIEW

**Comment by takagi** [comment]
Seems that `lanczos` and `kaiser_bessel_derived` windows are introduced since SciPy 1.10. Would you mark the tests to require newer SciPy?

**Comment by ev-br** [comment]
Done in [2e52b82](https://github.com/cupy/cupy/pull/8081/commits/2e52b820781e6d1b2ce97f73e2751cca7afc614d).

**Comment by takagi** [comment]
Thanks! I've kicked the CI again.

File: cupyx/scipy/signal/windows/_windows.py [hunk 1]
```diff
@@ -1424,6 +1424,91 @@ def kaiser(M, beta, sym=True):
...
+
+
+    References
+    ----------
+    .. [1] Bosi, Marina, and Richard E. Goldberg. Introduction to Digital
+           Audio Coding and Standards. Dordrecht: Kluwer, 2003.
+    .. [2] Wikipedia, "Kaiser window",
+           https://en.wikipedia.org/wiki/Kaiser_window
+
+    Examples
```
Comment (takagi) [inline cupyx/scipy/signal/windows/_windows.py:1468]: Would you fix this examples section to work with cupy? Or I suppose removing this section is also okay as users can refer to scipy's.

File: cupyx/scipy/signal/windows/_windows.py [hunk 2]
```diff
@@ -1424,6 +1424,91 @@ def kaiser(M, beta, sym=True):
     return _truncate(w, needs_trunc)
 
 
+def kaiser_bessel_derived(M, beta, sym=True):
```
Comment (takagi) [inline cupyx/scipy/signal/windows/_windows.py:1427]:
```suggestion
def kaiser_bessel_derived(M, beta, *, sym=True):
```

File: cupyx/scipy/signal/windows/_windows.py [hunk 3]
```diff
@@ -2036,6 +2121,64 @@ def taylor(M, nbar=4, sll=30, norm=True, sym=True):
     return _truncate(w, needs_trunc)
 
 
+def lanczos(M, sym=True):
```
Comment (takagi) [inline cupyx/scipy/signal/windows/_windows.py:2124]: To get aligned with SciPy.
```suggestion
def lanczos(M, *, sym=True):
```

File: cupyx/scipy/signal/windows/_windows.py (hunk 2 above)
Comment (ev-br) [reply cupyx/scipy/signal/windows/_windows.py:1427]: This breaks `get_window` which is not equipped to handle keyword-only arguments. Note that the bulk of the windowing APIs come from cuSignal, and probably predates both introduction of these two windows and the keywod-only stuff

File: cupyx/scipy/signal/windows/_windows.py (hunk 1 above)
Comment (ev-br) [reply cupyx/scipy/signal/windows/_windows.py]: Removed and added a `See Also`  link to the scipy version.

File: cupyx/scipy/signal/windows/_windows.py (hunk 2 above)
Comment (takagi) [reply cupyx/scipy/signal/windows/_windows.py:1427]: I got it. So lets leave the signatures not to have keyword-only parameters in this PR. Thanks!

**Review by takagi — APPROVED** [review]
LGTM!

**Pull request merged**